### PR TITLE
Add a TRACE log with TLS connection details on replication connections

### PR DESCRIPTION
**Pull request by sgmiller**
Logs some basic parameters of the TLS session when establishing a replication connection.

### CODE REVIEW

File: vault/cluster/cluster.go [hunk 1]
```diff
@@ -438,7 +440,23 @@ func (cl *Listener) GetDialerFunc(ctx context.Context, alpn string) func(string,
 		tlsConfig.NextProtos = []string{alpn}
 		cl.logger.Debug("creating rpc dialer", "address", addr, "alpn", alpn, "host", tlsConfig.ServerName)
 
-		return cl.networkLayer.Dial(addr, timeout, tlsConfig)
+		conn, err := cl.networkLayer.Dial(addr, timeout, tlsConfig)
+		if err != nil {
+			return nil, err
+		}
+		logTLSConnection(cl.logger, conn.ConnectionState())
```
Comment (briankassouf) [inline vault/cluster/cluster.go:447]: Can we wrap this in a `if cl.logger.IsTrace() {}` branch to avoid doing this extra work all the time?

Comment (sgmiller) [reply vault/cluster/cluster.go:447]: Good idea

**Review by briankassouf — APPROVED** [review]

**Comment by ncabatoff** [comment]
Feels like this could be very verbose, what about a new env var to gate this behaviour?

File: vault/cluster/cluster.go [hunk 2]
```diff
@@ -438,7 +440,25 @@ func (cl *Listener) GetDialerFunc(ctx context.Context, alpn string) func(string,
...
+		if cl.logger.IsTrace() {
+			logTLSConnection(cl.logger, conn.ConnectionState())
+		}
+		return conn, nil
+	}
+}
+
+func logTLSConnection(logger log.Logger, state tls.ConnectionState) {
+	cipherName, _ := tlsutil.GetCipherName(state.CipherSuite)
+	logger.Trace("TLS connection established", "negotiated_protocol", state.NegotiatedProtocol, "cipher_suite", cipherName)
```
[the remark this answers is not in the document]
Comment (sgmiller) [reply vault/cluster/cluster.go:456]: Is the addr ever sensitive?

**Comment by sgmiller** [comment]
It is a bit verbose, but those connections aren't made frequently are they?

File: vault/cluster/cluster.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ncabatoff) [reply vault/cluster/cluster.go:456]: Yes, but I think we already log that kind of info, and I tend to think of logs as less of a concern for sensitive data than metrics or things we return to the user.  Like we shouldn't put secrets, but I think it's okay to log addrs.  Of course, if we add an opt-in env var that makes it even less of a concern.

**Comment by ncabatoff** [comment]
> It is a bit verbose, but those connections aren't made frequently are they?

Not in the happy case, no.  But if things aren't working properly, it could be very frequent I think.

**Comment by sgmiller** [comment]
I'd have to imagine in an unhappy case lots of trace logs would be firing often. But I'm not opposed to gating this.

**Comment by sgmiller** [comment]
Added env gating and logging on the Accept side

**Pull request merged**